components/ntp_background_images/browser/ntp_background_images_source.cc

```diff
@@ -118,24 +139,52 @@ bool NTPBackgroundImagesSource::IsWallpaperPath(const std::string& path) const {
 }
 
 bool NTPBackgroundImagesSource::IsLogoPath(const std::string& path) const {
-  return path.compare(kLogoPath) == 0;
+  const std::string target_logo_path =
+      IsSuperReferralPath(path) ? std::string(kSuperReferralPath) + kLogoPath
```

Comment:
this also defeats the purpose of combining them because we can't create a new type or change anything without also updating the client. The client should only care about the metadata

Reply:
this can be changed in a follow up

Reply:
Currently, `chrome://branded-wallpaper/sponsored-images/...` is used for SI and `chrome://branded-wallpaper/super-referral/...` is used for SR.
`chrome://branded-wallpaper/super-referral/...` is used because we only use one additional SR for now. But, If want to use multiple themes at a time, I think it could be used like `chrome://branded-wallpaper/super-referral/theme_name/...` in the future.
and If we want to get default BGs via component , I think `chrome://branded-wallpaper/default/...` could be the base path for it.

Reply:
@bridiver As we discussed in DM, I think embedding component id into resource link would be good idea. I'll try it as a f/u.